packages/react-dom/src/__tests__/ReactTestUtilsAct-test.js

```diff
@@ -503,16 +503,15 @@ function runActTests(label, render, unmount, rerender) {
       // @gate __DEV__
       it('warns if you try to interleave multiple act calls', async () => {
         spyOnDevAndProd(console, 'error');
-        // let's try to cheat and spin off a 'thread' with an act call
-        (async () => {
-          await act(async () => {
-            await sleep(50);
-          });
-        })();
```

Comment:
This lead to an unhandled promise rejection and therefore uncaught exception in latest Node 16.x and 18.x. Jest did not considerd this test failed and instead exited. Though it seems to me that it should? /cc @SimenB 

New code results in no unhandled promise reject while still ensuring these act calls are interleaved.

Reply:
we should definitely pick up unhandled rejections - could you open up an issue so we at least track it?

Reply:
@SimenB  https://github.com/facebook/jest/issues/9210 seems like it's tracking the same issue

Reply:
good find, thanks!

Reply:
The test (before this PR) passes using jest-circus fwiw. So seems to be a jasmine problem